src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -137,14 +138,24 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
                         return (int)ReturnCode.ArgumentError;
                     }
 
-                    collectRundownEvents = selectedProfile.Rundown;
+                    rundownKeyword = selectedProfile.RundownKeyword;
+                    retryStrategy = selectedProfile.RetryStrategy;
 
                     Profile.MergeProfileAndProviders(selectedProfile, providerCollection, enabledBy);
                 }
 
                 if (rundown.HasValue)
                 {
-                    collectRundownEvents = rundown.Value;
+                    if (rundown.Value)
+                    {
+                        rundownKeyword |= EventPipeSession.DefaultRundownKeyword;
+                        retryStrategy = RetryStrategy.DropKeywordKeepRundown;
```

Comment:
If rundownKeyword was previously any subset of the DefaultRundownKeyword then it may trigger the asserts below. I think it would be easiest to just let an unneeded retry happen but if you are determined to detect and avoid it you can.